tests/nixos/gzip-content-encoding.nix

```diff
@@ -0,0 +1,66 @@
+# Test that doubly-compressed files fetched from server with double
+# compression. E.g. fetching a zstd compressed tarball from a server,
+# which compresses the response with `Content-Encoding: gzip`.
+
+{ lib, config, nixpkgs, ... }:
+
+let
+  pkgs = config.nodes.machine.nixpkgs.pkgs;
+
+  ztdCompressedFile = pkgs.stdenv.mkDerivation {
+    name = "dummy-zstd-compressed-archive";
+    dontUnpack = true;
+    nativeBuildInputs = with pkgs; [ zstd ];
+    buildPhase = ''
+      tar --zstd -cf archive.tar.zst ${nixpkgs}
```

Comment:
Can we use a small ad-hoc set of files here rather than `nixpkgs`? That could make a pretty big difference in the duration of the test